omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs1_9/EncounterSearchHandler1_9.java

```diff
@@ -0,0 +1,162 @@
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchConfig;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchHandler;
+import org.openmrs.module.webservices.rest.web.resource.api.SearchQuery;
+import org.openmrs.module.webservices.rest.web.resource.impl.EmptySearchResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.ConceptResource1_8;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8.PatientResource1_8;
+import org.openmrs.util.OpenmrsUtil;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+
+/**
+ * Allow searching for encounters based on an obs within the encounter, filtering by obs numeric,
+ * text, or coded values.
+ *
+ * Sample REST requests:
+ *
+ * 	Filter by coded value:
+ * 		encounter?s=byObs
+ * 		&patient=98bd219e-2e3a-41be-ae4e-bbd129d943d0
+ * 		&obsConcept=0dc64225-e9f6-11e4-a8a3-54ee7513a7ff
+ * 		&obsValues=3cd6f600-26fe-102b-80cb-0017a47871b2
+ *
+ * 	Filter by numeric values:
+ * 		encounter?s=byObs
+ *  * 		&patient=98bd219e-2e3a-41be-ae4e-bbd129d943d0
+ *  * 		&obsConcept=0dc64225-e9f6-11e4-a8a3-54ee7513a7ff
+ *  * 		&obsValues=150,175
+ *
+ */
+@Component
+public class EncounterSearchHandler1_9 implements SearchHandler {
+	
+	private static final String REQUEST_PARAM_PATIENT = "patient";
+	
+	private static final String REQUEST_PARAM_CONCEPT = "obsConcept";
+	
+	private static final String REQUEST_PARAM_VALUES = "obsValues";
+	
+	SearchQuery searchQuery = new SearchQuery.Builder("Allows you to search for encounters by observation values "
+	        + "given by a concept. Values are a comma delimited string of numeric, text or uuid values.")
+	        .withRequiredParameters(REQUEST_PARAM_PATIENT, REQUEST_PARAM_CONCEPT, REQUEST_PARAM_VALUES).build();
+	
+	private final SearchConfig searchConfig = new SearchConfig("byObs",
+	        RestConstants.VERSION_1 + "/encounter",
+	        Arrays.asList("1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*"),
+	        searchQuery);
+	
+	@Override
+	public SearchConfig getSearchConfig() {
+		return searchConfig;
+	}
+	
+	@Override
+	public PageableResult search(RequestContext context) throws ResponseException {
+		String patientUuid = context.getRequest().getParameter(REQUEST_PARAM_PATIENT);
+		String conceptUuid = context.getRequest().getParameter(REQUEST_PARAM_CONCEPT);
+		String values = context.getRequest().getParameter(REQUEST_PARAM_VALUES);
+		
+		if (StringUtils.isNotBlank(patientUuid) && StringUtils.isNotBlank(conceptUuid) && StringUtils.isNotBlank(values)) {
```

Comment:
I didn't think about initially, but let's add the ability to just pass in a patient and a list of concepts, in which case it doesn't filter by value.

Reply:
Makes sense @mogoodrich , I thought of that too. I've created a new PR that allows optional filtering on value.